src/solver/optimisation/adequacy_patch.cpp

```diff
@@ -26,121 +26,287 @@
     }
-    else
+    Interco = ProblemeHebdo->IndexDebutIntercoExtremite[Area];
+    while (Interco >= 0)
     {
-        Xmax = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
-        Xmin = -(ValeursDeNTC->ValeurDeNTCExtremiteVersOrigine[Interco]);
+        if (ProblemeHebdo->adequacyPatchRuntimeData.originAreaMode[Interco]
+            == physicalAreaInsideAdqPatch)
+        {
+            netPositionInit += ProblemeHebdo->ValeursDeNTC[hour]->ValeurDuFlux[Interco];
+        }
+        else if (ProblemeHebdo->adequacyPatchRuntimeData.originAreaMode[Interco]
+                 == physicalAreaOutsideAdqPatch)
+        {
+            flowsNode1toNodeA
+              += Math::Max(0.0, ProblemeHebdo->ValeursDeNTC[hour]->ValeurDuFlux[Interco]);
+        }
+        Interco = ProblemeHebdo->IndexSuivantIntercoExtremite[Interco];
     }
+
+    ensInit = ProblemeHebdo->ResultatsHoraires[Area]->ValeursHorairesDeDefaillancePositive[hour];
+    if (includeFlowsOutsideAdqPatchToDensNew)
+        densNew = Math::Max(0.0, ensInit + netPositionInit + flowsNode1toNodeA);
+    else
+        densNew = Math::Max(0.0, ensInit + netPositionInit);
+
+    return std::make_pair(netPositionInit, densNew);
 }
 
-void setBoundsNoAdqPatch(double& Xmax,
-                         double& Xmin,
-                         VALEURS_DE_NTC_ET_RESISTANCES* ValeursDeNTC,
-                         const int Interco)
+template<class T>
+void sumTwoArrays(T* a, T* b, int num)
 {
-    Xmax = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
-    Xmin = -(ValeursDeNTC->ValeurDeNTCExtremiteVersOrigine[Interco]);
+    for (uint i = 0; i < num; ++i)
+        a[i] += b[i];
+}
+template void sumTwoArrays<double>(double* a, double* b, int num);
+
+void HOURLY_CSR_PROBLEM::resetProblem()
+{
+    OPT_LiberationProblemesSimplexe(pWeeklyProblemBelongedTo);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemVariables()
+{
+    OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo,
+                                                                         *this);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemConstraintsLHS()
+{
+    OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::setVariableBounds()
+{
+    OPT_InitialiserLesBornesDesVariablesDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::buildProblemConstraintsRHS()
+{
+    OPT_InitialiserLeSecondMembreDuProblemeQuadratique_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::setProblemCost()
+{
+    OPT_InitialiserLesCoutsQuadratiques_CSR(pWeeklyProblemBelongedTo, *this);
+}
+
+void HOURLY_CSR_PROBLEM::solveProblem()
+{
+    OPT_AppelDuSolveurQuadratique_CSR(pWeeklyProblemBelongedTo->ProblemeAResoudre, *this);
+}
+
+void HOURLY_CSR_PROBLEM::run()
+{
+    resetProblem();
+    calculateCsrParameters(pWeeklyProblemBelongedTo, *this);
```

Comment:
Is it necessary to pass `*this`, consider making this function a member-function.